feat: add `math/base/special/cphasef`

anandkaranubc opened the pull request:
Progresses https://github.com/stdlib-js/stdlib/issues/649

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the complete implementation for `math/base/special/cphasef`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: add `math/base/special/cphasef`

anandkaranubc committed:
Merge branch 'develop' into feat/cphasef

anandkaranubc committed:
test: update tolerances

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: fix header include

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/cphasef/README.md:
```diff
@@ -0,0 +1,202 @@
+        v = x[ i ];
+        y = stdlib_base_cphasef( v );
+        stdlib_complex64_reim( v, &re, &im );
+        printf( "fcphasef(%f + %fi) = %f\n", re, im, y );
```
Was this supposed to be

```suggestion
        printf( "cphasef(%f + %fi) = %f\n", re, im, y );
```

(also notice that JS example above has `arg`)?

Same question applies to `example.c` file.

anandkaranubc replied:
That was being used in `cphase`, but I can revisit that and update it too.

anandkaranubc replied:
https://github.com/stdlib-js/stdlib/blob/0160d55258f7e58e3e9bb2ad1a2c345575343515/lib/node_modules/%40stdlib/math/base/special/cphase/examples/index.js#L30

anandkaranubc replied:
https://github.com/stdlib-js/stdlib/blob/0160d55258f7e58e3e9bb2ad1a2c345575343515/lib/node_modules/%40stdlib/math/base/special/cphase/examples/c/example.c#L41

Planeshifter replied:
I think that has to be a typo. The `README.md` of `cphase` uses `cphase`:

https://github.com/stdlib-js/stdlib/tree/0160d55258f7e58e3e9bb2ad1a2c345575343515/lib/node_modules/%40stdlib/math/base/special/cphase

anandkaranubc replied:
Should I also move forward and do the same changes in `cphase` as done here?

Planeshifter replied:
Yes, that would be great!

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/cphasef/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,138 @@
+
+	for ( i = 0; i < 100; i++ ) {
+		re[ i ] = ( 1000.0f*rand_float() ) - 500.0f;
+		im[ i ] = ( 1000.0f*rand_float() ) - 500.0f;
```
Can we generate the numbers via a `random_uniform` function, please?

anandkaranubc replied:
On it!

anandkaranubc committed:
Merge branch 'develop' into feat/cphasef

anandkaranubc committed:
bench: use `random_uniform`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: refactor examples to use `log-each-map`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented:
I used `log-each-map` for the JS examples to be consistent with the other examples. https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/cinvf/examples/index.js

Planeshifter reviewed (approved):
Let's get it in! Thanks, Karan!
